### ISSUE TITLE
Feature resolution is unexpectedly impacted by the size of the bounding box

### ISSUE THREAD

**Issue by Wulfsta** [libfive/libfive#562]
### Description

Resolution is impacted when changing the size of the bounding box.

#### Steps to Reproduce

Swap the commented `set_bounds` call in the following Studio script:
```
import studio

### SWAP THESE TO OBSERVE IMPACT ###
studio.set_bounds([-10, -10, -10], [10, 10, 10])
#studio.set_bounds([-10, -10, -1], [10, 10, 6])

test_quality = 9
test_resolution = 10

studio.set_quality(test_quality)
studio.set_resolution(test_resolution)

from libfive.stdlib import *

def quality_test():
    norm = Shape.sqrt(Shape.X() * Shape.X() + Shape.Y() * Shape.Y())
    return Shape.nanfill(Shape.sin(1000.0 / (norm)), 1.0)

qt = intersection(
    quality_test(),
    cylinder(9.9, 5)
)

qt
```

Notice that in the first case there are 11 rings generated before artifacts appear, while in the second case there are only three rings. I would not anticipate this behavior as an end user, and it took me a while to realize that it was not the quality or resolution that was causing artifacts in one of my designs.

### Versions

- Operating system: NixOS (Linux)
- `libfive` commit hash: 7af5f43684a8a497ac8610d39f7fca935364a9b9
  - Is this the latest commit to `master`?  yes
- Qt version (if applicable): 6.6.2

**Comment by mkeeter** [comment]
This is a somewhat confusing side-effect of how we pick the max subdivision level in `region.hpp`:

```cpp
    /*
     *  Returns a version of this region that has the level set
     *  based on the given minimum feature.
     *
     *  This lets us do subdivision without worrying that the termination
     *  condition (of a cell side being < min_feature) is dependent on
     *  floating-point accuracy.
     */
    Region<N> withResolution(double min_feature) const {
        const auto min_dimension = (upper - lower).minCoeff();
        const auto level = ceil(fmaxf(0.0f, logf(min_dimension / min_feature)) /
                                logf(2));
        return Region<N>(lower, upper, perp, level);
    }
```

The number of octree subdivisions is set by the lowest dimension, so I think that going from [10, 10] → [-1, 6] must be reducing it in your example.

Maybe it should be set by the largest dimension instead?

**Comment by Wulfsta** [comment]
I agree that this should be per unit, and not dependent on the size of the bounding box. It also has a huge impact on exported STL size - settings that produce no artifacts with a smaller bounding box produce significantly larger files (multiple GiBs in my case), whereas a bounding cube produces no artifacts at default settings and order of magnitude smaller files.

**Comment by junpls** [comment]
I can confirm this issue from my observations using the [Unity bindings](https://github.com/zalo/libfive-unity) in combination with libfive binaries that I compiled from [e837098](https://github.com/libfive/libfive/commit/e8370983e7bc6d49409affcc34fc70c673cc876f) (latest `master`).

This scene contains nothing but a cube with a side length of 1.5 (shifted up slightly) subtracted from a cube with a side length of 2.
As I tweak nothing but the bounds size, vastly different meshes are generated:

<img width="1246" height="768" alt="Image" src="https://github.com/user-attachments/assets/a2b1266e-9b0b-49a2-bd89-175a0ff82326" />
<img width="1246" height="768" alt="Image" src="https://github.com/user-attachments/assets/9f842187-a6d6-4269-ad5b-8ea0f72f8ce7" />
<img width="1246" height="768" alt="Image" src="https://github.com/user-attachments/assets/9e219385-8459-4ef2-afc8-96eb59b50da1" />
<img width="1246" height="768" alt="Image" src="https://github.com/user-attachments/assets/b12deec6-a43b-400c-a576-e81f8aeb43c2" />


Even with only the 2x2x2 cube, certain numbers for the bounds can produce overly detailed meshes:

<img width="1246" height="768" alt="Image" src="https://github.com/user-attachments/assets/55a07bfc-6a8b-4dd5-92af-c8aee275b503" />
<img width="1246" height="768" alt="Image" src="https://github.com/user-attachments/assets/1bd22244-e186-477f-9804-accb774bc052" />

**Comment by Wulfsta** [comment]
@mkeeter out of curiosity, does Fidget exhibit this behavior as well? As a suggestion, it would be nice to define subregions with different resolutions if possible - if overall a part only needs a loose tolerance, but particular regions need a tighter one this would be useful.

**Comment by junpls** [comment]
I found another example, which is reproducible in Studio:

```scheme
;; Try these different bounds. `2.0` will produce a broken mesh.
;; Also, `2.1` will produce a notably more detailed mesh than `1.9`.
;(define bounds 1.9)
(define bounds 2.0)
;(define bounds 2.1)

(set-bounds! [(- bounds) (- bounds) (- bounds)] [bounds bounds bounds])
(set-quality! 8)
(set-resolution! 8)

(difference
  (cube [-0.725 -0.725 -0.725] [0.725 0.725 0.725])
  (sphere 0.9))
```

<img width="1119" height="375" alt="Image" src="https://github.com/user-attachments/assets/fe06c8ad-471a-49b8-9e78-0f9cb4ed5d85" />

Maybe out of curiosity: Is there some kind of formula by which one could calculate "ideal bounds" based on the geometry in the scene and the resolution? I know that's not a fix, but I was hoping I could use libfive for a user-facing level editing feature in a game, meaning that triangle count matters and I can't expect the user to tweak the bounds manually if something goes unintended.
